main.go

```diff
@@ -50,37 +52,57 @@ func process(typeName string, fileName string, packageName string, mode string)
 		return err
 	}
 
-	f, err := parser.ParseFile(token.NewFileSet(), fileName, inputCode, parser.ParseComments)
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, fileName, inputCode, parser.ParseComments)
 	if err != nil {
 		return err
 	}
 
+	expectedLine, _ := strconv.Atoi(lineNum)
+	expectedLine += 1
+
 	var specs [][2]string
 
-	ast.Inspect(f, func(node ast.Node) bool {
-		spec, ok := node.(*ast.ValueSpec)
-		if !ok {
+	ast.Inspect(f, func(astNode ast.Node) bool {
+		node, ok := astNode.(*ast.GenDecl)
+		if !ok || (node.Tok != token.CONST && node.Tok != token.VAR) {
 			return true
 		}
 
-		if len(spec.Names) != 1 {
+		position := fset.Position(node.Pos())
+		if position.Line != expectedLine {
 			return false
```

Comment:
guess you need true here. we did not find position in AST yet, so it may or may not be inside subtree. good to iterate further. this is akin to continuing tree walking until detection of var/const declaration above. (can't think example now yet though).
```suggestion
			return true
```

Reply:
I believe false is correct here, since at this point we know already know we are in const/var block, so if location in the file does not match then there is not point in continuing with children. The traversal will retreat one level up and proceed from there.

Reply:
right.